acvm-repo/acvm/tests/solver.rs

```diff
@@ -827,6 +1065,31 @@ fn prop_assert_zero_l(
     (solve_blackbox_func_call(op, op_zero, x), FieldElement::zero())
 }
 
+// Test that varying one of the inputs produces a different result
+//
+// (is the op injective for the given inputs?, failure string)
+fn prop_assert_injective<F>(
+    inputs: Vec<ConstantOrWitness>,
+    distinct_inputs: Vec<ConstantOrWitness>,
+    num_outputs: usize,
+    op: F,
+) -> (bool, String)
```

Comment:
It's intentionally a `(bool, String)` to match the `prop_assert!` macro inputs. There's something curious going on with the `prop_assert..!` macros, because attempting to use them with type `A` says they're type `B` and vice versa. (The specific `A` and `B` are something like `()` and `Result<_, Some_proptest_error>`)